src/read.js

```diff
@@ -1,10 +1,57 @@
 "use strict";
 
+const { TextDecoder, TextEncoder } = require("text-encoding");
+
 var copychars = require("@dcos/copychars").default;
 
 var RECORD_PATTERN = /^\d+\n.+/;
 
-module.exports = function read(input) {
+/**
+ *
+ * @param {Uint8Array} byteArray byteArray to parse
+ * @param {Uint8Array[]} records aray containing complete records
+ * @returns {Uint8Array} incomplete record data, if existent
+ */
+function parseRecordsWithRest(byteArray, records) {
+  if (byteArray.length === 0) {
+    return new Uint8Array();
+  }
+
+  const newlineIndex = byteArray.findIndex(charCode => charCode == 10);
```

Comment:
might be worth making a constant for `10` and maybe `48` below so its clear what they represent?

Reply:
yeah, good idea!